src/storage/workspace_storage.rs

```diff
@@ -180,80 +176,87 @@ impl WorkspaceStorage {
         manifest: PyObject,
         cache_only: bool,
         removed_ids: Option<HashSet<ChunkID>>,
-    ) -> PyResult<FutureIntoCoroutine> {
-        let ws = self.get_storage()?;
-        let manifest = file_or_folder_manifest_from_py_object(py, &manifest)?;
-        let removed_ids = removed_ids.map(|x| {
-            x.into_iter()
-                .map(|id| libparsec::core_fs::ChunkOrBlockID::ChunkID(id.0))
-                .collect()
-        });
-
-        let fut = if cache_only {
-            FutureIntoCoroutine::ready(
-                ws.set_manifest_in_cache(entry_id.0, manifest, removed_ids)
-                    .map_err(fs_to_python_error)
-                    .map(|_| py.None()),
-            )
-        } else {
-            FutureIntoCoroutine::from(async move {
-                ws.set_manifest(entry_id.0, manifest, cache_only, removed_ids)
-                    .await
-                    .map_err(fs_to_python_error)
-            })
-        };
-
-        Ok(fut)
+    ) -> FutureIntoCoroutine {
+        match self.get_storage() {
+            Ok(ws) => match file_or_folder_manifest_from_py_object(py, &manifest) {
```

Comment:
```suggestion
        match self.get_storage().and_then(|ws| file_or_folder_manifest_from_py_object(py, &manifest).map(|manifest| (ws, manifest)))
            Ok(ws) => match file_or_folder_manifest_from_py_object(py, &manifest) {
```

We could merge both `get_storage` and `file_or_folder_manifest_...` this would remove 1 layer of indentation

Reply:
:+1:

Reply:
Done